services/venice-controller/src/test/java/com/linkedin/venice/controller/grpc/server/ClusterAdminOpsGrpcServiceImplTest.java

```diff
@@ -148,18 +148,33 @@ public void testGetAdminTopicMetadataSuccess() {
 
   @Test
   public void testUpdateAdminTopicMetadataSuccess() {
-    UpdateAdminTopicMetadataGrpcResponse response =
-        UpdateAdminTopicMetadataGrpcResponse.newBuilder().setClusterName(TEST_CLUSTER).build();
+    AdminTopicGrpcMetadata.Builder adminTopicGrpcMetadataBuilder = AdminTopicGrpcMetadata.newBuilder()
+        .setClusterName(TEST_CLUSTER)
+        .setExecutionId(EXECUTION_ID)
+        .setOffset(100L)
+        .setUpstreamOffset(-1L);
+    AdminTopicMetadataGrpcResponse response =
+        AdminTopicMetadataGrpcResponse.newBuilder().setMetadata(adminTopicGrpcMetadataBuilder.build()).build();
     doReturn(response).when(requestHandler).updateAdminTopicMetadata(any(UpdateAdminTopicMetadataGrpcRequest.class));
     doReturn(true).when(accessManager).isAllowListUser(anyString(), any());
 
     UpdateAdminTopicMetadataGrpcRequest request = UpdateAdminTopicMetadataGrpcRequest.newBuilder()
-        .setMetadata(AdminTopicGrpcMetadata.newBuilder().setClusterName(TEST_CLUSTER).setExecutionId(EXECUTION_ID))
+        .setMetadata(
```

Comment:
Why do we need to change/update request?

Reply:
I think it doesn't represent a valid request since we required the offsets to be provided if no store is provided.

```
    if (storeName != null && (offset != null || upstreamOffset != null)) {
      throw new VeniceException("Updating offsets is not allowed for store-level admin topic metadata");
    } else if (storeName == null && (offset == null || upstreamOffset == null)) {
      throw new VeniceException("Offsets must be provided to update cluster-level admin topic metadata");
    }
```

Reply:
If that's the case existing test should be failing with an exception, no?

Reply:
The test is not failing since the test is not to test the logic since the response is mocked, it is testing grpc end. Aligned offline to keep this.